py/modmicropython.c

```diff
@@ -166,6 +167,18 @@ static mp_obj_t mp_micropython_schedule(mp_obj_t function, mp_obj_t arg) {
 static MP_DEFINE_CONST_FUN_OBJ_2(mp_micropython_schedule_obj, mp_micropython_schedule);
 #endif
 
+#if MICROPY_HELPER_REPL
+static mp_obj_t mp_micropython_repl_autocomplete(mp_obj_t cur_line) {
+    const char *compl_str;
+    size_t str_len = 0;
+    const char *str = mp_obj_str_get_data(cur_line, &str_len);
+
+    ssize_t compl_len = mp_repl_autocomplete(str, str_len, &mp_plat_print, &compl_str);
+    return (compl_len <= 0) ? mp_const_none : mp_obj_new_str_via_qstr(compl_str, compl_len);
```

Comment:
If you make this `(compl_len < 0)` then the caller should be able to distinguish between the 3 distinct cases (no match, one match, many matches).